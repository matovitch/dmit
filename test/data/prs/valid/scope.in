func f()
{
    let i : i64;
    i = 3;

    {
        let i : f32;
        i = 3.14;
    }
    {}
    {
        { x = y; }
        { return 42; }
    }
}
