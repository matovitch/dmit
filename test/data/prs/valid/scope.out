{"tree" :{"nodes":[{"kind":"END_OF_TREE","size":0,"childCount":0,"start":0,"stop":0},{"kind":"IDENTIFIER","size":0,"childCount":0,"start":71,"stop":70},{"kind":"FUN_ARGUMENTS","size":0,"childCount":0,"start":70,"stop":67},{"kind":"IDENTIFIER","size":0,"childCount":0,"start":63,"stop":61},{"kind":"IDENTIFIER","size":0,"childCount":0,"start":59,"stop":58},{"kind":"DCL_VARIABLE","size":2,"childCount":2,"start":65,"stop":58},{"kind":"IDENTIFIER","size":0,"childCount":0,"start":56,"stop":54},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":54,"stop":52},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":52,"stop":51},{"kind":"EXP_ASSIGN","size":3,"childCount":3,"start":56,"stop":51},{"kind":"IDENTIFIER","size":0,"childCount":0,"start":45,"stop":43},{"kind":"IDENTIFIER","size":0,"childCount":0,"start":41,"stop":40},{"kind":"DCL_VARIABLE","size":2,"childCount":2,"start":47,"stop":40},{"kind":"IDENTIFIER","size":0,"childCount":0,"start":38,"stop":36},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":36,"stop":34},{"kind":"LIT_DECIMAL","size":0,"childCount":0,"start":34,"stop":33},{"kind":"EXP_ASSIGN","size":3,"childCount":3,"start":38,"stop":33},{"kind":"SCOPE","size":7,"childCount":2,"start":49,"stop":29},{"kind":"SCOPE","size":0,"childCount":0,"start":29,"stop":26},{"kind":"IDENTIFIER","size":0,"childCount":0,"start":22,"stop":20},{"kind":"EXP_OPERATOR","size":0,"childCount":0,"start":20,"stop":18},{"kind":"IDENTIFIER","size":0,"childCount":0,"start":18,"stop":17},{"kind":"EXP_ASSIGN","size":3,"childCount":3,"start":22,"stop":17},{"kind":"SCOPE","size":4,"childCount":1,"start":24,"stop":13},{"kind":"LIT_INTEGER","size":0,"childCount":0,"start":9,"stop":8},{"kind":"STM_RETURN","size":1,"childCount":1,"start":11,"stop":8},{"kind":"SCOPE","size":2,"childCount":1,"start":13,"stop":4},{"kind":"SCOPE","size":8,"childCount":2,"start":26,"stop":2},{"kind":"SCOPE","size":25,"childCount":5,"start":67,"stop":0},{"kind":"DEF_FUNCTION","size":28,"childCount":3,"start":73,"stop":0},{"kind":"DEFINITION","size":29,"childCount":1,"start":73,"stop":0},{"kind":"MODULE","size":30,"childCount":1,"start":73,"stop":0}]},"errors":[{"offset":4294967295,"errors":[]}]}